diff -Naur poco-1.7.7-all.orig/Foundation/include/Poco/LogFile_STD.h poco-1.7.7-all/Foundation/include/Poco/LogFile_STD.h 
--- poco-1.7.7-all.orig/Foundation/include/Poco/LogFile_STD.h	2016-12-31 01:58:53.000000000 -0600
+++ poco-1.7.7-all/Foundation/include/Poco/LogFile_STD.h	2017-04-07 18:40:23.790899248 -0500
@@ -45,6 +45,7 @@
 	std::string _path;
 	mutable Poco::FileOutputStream _str;
 	Timestamp _creationDate;
+	UInt64 _size;
 };

