diff -Naur poco-1.7.7-all.orig/Foundation/src/LogFile_STD.cpp poco-1.7.7-all/Foundation/src/LogFile_STD.cpp 
--- poco-1.7.7-all.orig/Foundation/src/LogFile_STD.cpp	2016-12-31 01:58:53.000000000 -0600
+++ poco-1.7.7-all/Foundation/src/LogFile_STD.cpp	2017-04-09 18:40:26.345159548 -0500
@@ -24,9 +24,10 @@
 
 LogFileImpl::LogFileImpl(const std::string& path): 
 	_path(path),
-	_str(_path, std::ios::app)
+	_str(_path, std::ios::app),
+	_size((UInt64) _str.tellp())
 {
-	if (sizeImpl() == 0)
+	if (_size == 0)
 		_creationDate = File(path).getLastModified();
 	else
 		_creationDate = File(path).created();
@@ -40,18 +41,25 @@
 
 void LogFileImpl::writeImpl(const std::string& text, bool flush)
 {
+	if (!_str.good())
+ 	{
+ 		_str.close();
+ 		_str.open(_path, std::ios::app);
+ 	}
+ 	if (!_str.good()) throw WriteFileException(_path);
 	_str << text;
 	if (flush)
 		_str << std::endl;
 	else
 		_str << "\n";
 	if (!_str.good()) throw WriteFileException(_path);
+	_size = (UInt64) _str.tellp();
 }
 
 
 UInt64 LogFileImpl::sizeImpl() const
 {
-	return (UInt64) _str.tellp();
+	return _size;
 }
